sdcm/cluster.py

```diff
@@ -2596,35 +2596,30 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
         :param publish_event: publish event or not
         :return: Remoter result object
         """
-        def _publish_event(subtype, severity, duration=None, error=None, full_traceback=None):
-            if publish_event:
-                NodetoolEvent(type=sub_cmd, subtype=subtype, severity=severity, node=self.name,
-                              duration=duration, options=options, error=error,
-                              full_traceback=full_traceback).publish()
-
         cmd = self._gen_nodetool_cmd(sub_cmd, args, options)
-        start_time = time.time()
-        try:
-            _publish_event(subtype='start', severity=Severity.NORMAL)
 
-            result = self.remoter.run(cmd, timeout=timeout, ignore_status=ignore_status, verbose=verbose)
-            self.log.debug("Command '%s' duration -> %s s" % (result.command, result.duration))
+        with NodetoolEvent(nodetool_command=sub_cmd,
+                           node=self.name,
+                           options=options,
+                           publish_event=publish_event) as nodetool_event:
+            try:
+                result = self.remoter.run(cmd, timeout=timeout, ignore_status=ignore_status, verbose=verbose)
+                self.log.debug("Command '%s' duration -> %s s" % (result.command, result.duration))
 
-            _publish_event(subtype='end', severity=Severity.NORMAL, duration=f"{result.duration}s")
-            return result
-        except Exception as details:  # pylint: disable=broad-except
-            event_severity_on_failure = Severity.ERROR
-            if warning_event_on_exception and isinstance(details, warning_event_on_exception):
-                event_severity_on_failure = Severity.WARNING
+                nodetool_event.duration = result.duration
+                return result
+            except Exception as details:  # pylint: disable=broad-except
+                event_severity_on_failure = Severity.ERROR
```

Comment:
no need to set the error event severity since it is default, set warning only

Reply:
fixed